lasy/laser.py

```diff
@@ -237,33 +244,76 @@ def propagate(self, distance, nr_boundary=None, backend="NP", show_progress=True
+                spatial_axes_n = (grid.axes[0],)  # resampled axis
+                # Overwrite grid with new grid (for the resampled field)
+                self.grid = grid
+                # Creating an empty array to store the resampled spectral field.
+                # This will be needed in the propagation step below.
+                spectral_field_n = np.zeros(
+                    (len(self.grid.azimuthal_modes), grid.npoints[0], grid.npoints[1]),
+                    dtype="complex128",
+                )
+                self.prop = []  # Delete existing propagator
+                # Construct the propagator and pass resampled axis
                 for m in self.grid.azimuthal_modes:
                     self.prop.append(
                         PropagatorResampling(
                             *spatial_axes,
                             self.omega_1d / c,
+                            *spatial_axes_n,
                             mode=m,
                             backend=backend,
                             verbose=False,
                         )
                     )
-            # Propagate the spectral image
-            spectral_field = self.grid.get_spectral_field()
+            # No resampling (propagating on existing grid)
+            else:
+                if not hasattr(self, "prop"):
+                    spatial_axes = (self.grid.axes[0],)
+                    self.prop = []
+                    # Construct the propagator
+                    for m in self.grid.azimuthal_modes:
+                        self.prop.append(
+                            PropagatorResampling(
+                                *spatial_axes,
+                                self.omega_1d / c,
+                                mode=m,
+                                backend=backend,
+                                verbose=False,
+                            )
+                        )
+            # Propagate the spectral image in time
             for i_m in range(self.grid.azimuthal_modes.size):
                 transform_data = np.transpose(spectral_field[i_m]).copy()
-                self.prop[i_m].step(
+                # Propagation step. The spectral field is explicitely
+                # overwritten with the updated field.
+                transform_data = self.prop[i_m].step(
                     transform_data,
                     distance,
-                    overwrite=True,
+                    overwrite=False,
                     show_progress=show_progress,
                 )
-                spectral_field[i_m, :, :] = np.transpose(transform_data).copy()
-            self.grid.set_spectral_field(spectral_field)
+                if grid is not None:
+                    # Store the updated and resampled field.
+                    spectral_field_n[i_m, :, :] = np.transpose(transform_data).copy()
+                    spectral_field = spectral_field_n
```

Comment:
Could this be simplified to:
```suggestion
                    spectral_field[i_m, :, :] = np.transpose(transform_data).copy()
```